File: SENTINEL/services/shared-python/sentinel_streams/tenant_router.py
class TenantStreamRouter:
    def __init__(self, tenant_slug: str):
        self._slug = tenant_slug

    def resolve(self, stream: str) -> str:
        if not self._slug or self._slug == "default":
            return stream
        return f"{self._slug}.{stream}"

    def resolve_group(self, group: str) -> str:
        if not self._slug or self._slug == "default":
            return group
        return f"{self._slug}-{group}"

    @property
    def slug(self) -> str:
        return self._slug

File: SENTINEL/packages/stream-client/src/tenant-router.ts
/**
 * Routes stream names through a tenant namespace.
 * Stream names become: {tenantSlug}.{originalStream}
 * e.g., "default.sentinel.mempool.pending"
 *
 * When tenantSlug is "default" or empty, uses the original stream name
 * for backward compatibility with single-tenant mode.
 */
export class TenantStreamRouter {
    private tenantSlug: string;

    constructor(tenantSlug: string) {
        this.tenantSlug = tenantSlug;
    }

    /** Resolve a logical stream name to a tenant-scoped physical stream name */
    resolve(stream: string): string {
        if (!this.tenantSlug || this.tenantSlug === "default") {
            return stream; // backward-compatible: default tenant uses bare stream names
        }
        return `${this.tenantSlug}.${stream}`;
    }

    /** Resolve a consumer group name to include tenant scope */
    resolveGroup(group: string): string {
        if (!this.tenantSlug || this.tenantSlug === "default") {
            return group;
        }
        return `${this.tenantSlug}-${group}`;
    }

    get slug(): string {
        return this.tenantSlug;
    }
}
